### PR TITLE
Global Exception Handlers implementation

### PR DESCRIPTION
**Pull request by matt-jenner**
and tests

### CODE REVIEW

File: Just.Anarchy.Test.Unit/Actions/AnarchyActionFactoryTests.cs [hunk 1]
```diff
@@ -1,182 +1,182 @@
...
+            var action = Substitute.For<ICauseAnarchy>();
+            var timer = Substitute.For<IHandleTime>();
+            var sut = new AnarchyActionFactory(action, timer);
+            sut.ForTargetPattern("/bob$");
+
+            //Act
+            sut.HandleRequest(url);
+
+            //Assert
+            if (expMatch) {
```
Comment (luisrtp) [inline Just.Anarchy.Test.Unit/Actions/AnarchyActionFactoryTests.cs]: Do you think its worth splitting this into two separate tests? So we don't need a conditional.

File: Just.Anarchy.Test.Unit/Actions/AnarchyActionFactoryTests.cs [hunk 2]
```diff
@@ -1,182 +1,182 @@
...
+        [TestCase("/bob")]
+        [TestCase("")]
+        public void HandleRequestRejectsAllUrlsIfTargetPatternIsNull(string url)
+        {
+            //Arrange
+            var action = Substitute.For<ICauseAnarchy>();
+            var timer = Substitute.For<IHandleTime>();
+            var sut = new AnarchyActionFactory(action, timer);
+            sut.ForTargetPattern(null);
+            //Act
```
Comment (luisrtp) [inline Just.Anarchy.Test.Unit/Actions/AnarchyActionFactoryTests.cs:106]: Space? - Minor

File: Just.Anarchy.Test.Unit/Exceptions/Handlers/ExceptionHandlerManagerTests.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+            //Act
+            var result = Assert.CatchAsync(async () => await sut.HandleAsync(response, exception));
+
+            //Assert
+            result.Should().Be(exception);
+        }
+
+        [Test]
+        public void HandleAsyncChecksAllHandlers()
```
Comment (luisrtp) [inline Just.Anarchy.Test.Unit/Exceptions/Handlers/ExceptionHandlerManagerTests.cs]: This test and the above share the same setup, is it worth just moving the assert from line 66 to this test. I know we should keep assertions to a minimal, but this seems to be testing the same action.

File: Just.Anarchy.Test.Unit/utils/ObjectSerializationExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Just.Anarchy.Test.Unit.utils
+{
+    public static class ObjectSerializationExtensions
```
Comment (luisrtp) [inline Just.Anarchy.Test.Unit/utils/ObjectSerializationExtensions.cs:9]: Nice

File: Just.Anarchy.Test.Unit/utils/StreamExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace Just.Anarchy.Test.Unit.utils
+{
+    public static class StreamExtensions
+    {
+        public static string ConvertToString(this Stream stream)
```
Comment (luisrtp) [inline Just.Anarchy.Test.Unit/utils/StreamExtensions.cs:10]: Nice

**Review by luisrtp — APPROVED** [review]
A few minor comments, but nothing major or urgent.

File: Just.Anarchy.Test.Unit/Actions/AnarchyActionFactoryTests.cs (hunk 1 above)
Comment (matt-jenner) [reply Just.Anarchy.Test.Unit/Actions/AnarchyActionFactoryTests.cs]: Fair point, originally I think I was going to have more cases but pared it down again, splitting into two tests.

File: Just.Anarchy.Test.Unit/Exceptions/Handlers/ExceptionHandlerManagerTests.cs (hunk 3 above)
Comment (matt-jenner) [reply Just.Anarchy.Test.Unit/Exceptions/Handlers/ExceptionHandlerManagerTests.cs]: Yeah tried to keep single assertion per test ideally but I had thought the same myself.

**Pull request merged**